src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -74,22 +80,50 @@ public async Task<CommandOrCodeActionContainer> Handle(CodeActionParams request,
                 throw new ArgumentNullException(nameof(request));
             }
 
-            var razorCodeActions = await GetCodeActionsAsync(request, cancellationToken).ConfigureAwait(false);
-            if (razorCodeActions is null)
+            var razorCodeActionContext = await GenerateRazorCodeActionContextAsync(request, cancellationToken).ConfigureAwait(false);
+            if (razorCodeActionContext is null)
+            {
+                return null;
+            }
+
+            if (cancellationToken.IsCancellationRequested)
+            {
+               return null;
```

Comment:
Why return null instead of throwing a task canceled exception?

Reply:
Probably best to do task cancelled. Wanted to return null to avoid unhandled task cancellation exceptions causing external issues, but that really shouldn't be necessary.